### ISSUE TITLE
drush_backend_batch_process() should clear batch_get()

### ISSUE THREAD

**Issue by pfrenssen** [drush-ops/drush#3773]
**Describe the bug**
This is coming out of a bug report that was filed for Search API: [Issue #3011340: Multiple index update causes drush config import to get stuck](https://www.drupal.org/project/search_api/issues/3011340).

If a batch operation is added with `batch_set()` and then processed using `drush_backend_batch_process()` then the list of batch operations is not cleared. If a second call to `drush_backend_batch_process()` is made in the same request, then the original set of batch operations is requeued, leading to unexpected results.

**To Reproduce**

```
// Define a batch operation.
$batch_definition = [
  'operations' => [[[$this, 'processBatch'], [$task_ids, $conditions]]],
];
batch_set($batch_definition);

// This call will execute the batch operation.
drush_backend_batch_process();

// Define a second batch operation.
$batch_definition = [
  'operations' => [[[$this, 'processBatch'], [$task_ids, $conditions]]],
];
batch_set($batch_definition);

// This call will execute both the first and the second batch operation.
// Since the first one has already been processed this might cause unexpected results.
drush_backend_batch_process();
```

**Expected behavior**
`drush_backend_batch_process()` should clean up `batch_get()` after it is done.

**Workaround**

```
$result = drush_backend_batch_process();
if ($result['context']['drush_batch_process_finished'] === TRUE) {
  $batch = &batch_get();
  $batch = NULL;
}
```

### System Configuration
| Q               | A
| --------------- | ---
| Drush version?  | master
| Drupal version? | 8.x-dev
| PHP version     | 7.2
| OS?             | Linux

**Comment by weitzman** [comment]
I guess the code below is not sufficient? If we need more code tahts probably where it should go https://github.com/drush-ops/drush/blob/eaf9c3e4270dc1c8035d84e00e17935ee4a0ee9e/includes/batch.inc#L345-L349

**Comment by batkor** [comment]
subscribe
Thanks @pfrenssen

**Comment by lpeabody** [comment]
Just wanted to include this as a potentially related issue https://www.drupal.org/node/3068196, piggybacked from https://www.drupal.org/project/search_api/issues/3019652#comment-13167514. It provides additional context around the discussion @pfrenssen posted in the description.

**Comment by plach79** [comment]
Posted a PR implementing @pfrenssen's suggestion at https://github.com/drush-ops/drush/pull/4886

**Comment by weitzman** [comment]
Fixed in #4886
